File: 0274-h-index/0274-h-index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        sort(citations.begin(), citations.end(), std::greater<int>());
        int index = 0;
        for (int i = 0; i < citations.size(); i++) {
            if (citations[i] >= i + 1 ) index++;
            else break;
        }
        return index;
    }
};

File: 0274-h-index/0274-h-index.rs
impl Solution {
    pub fn h_index(citations: Vec<i32>) -> i32 {
        let mut citations = citations;
        citations.sort_by(|a, b| b.cmp(a));
        let mut index = 0;
        for i in 0..citations.len() {
            if citations[i] as usize >= i + 1 {
                index += 1;
            } else {
                break;
            }
        }
        index
    }
}